src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -456,97 +444,105 @@ private static void PerformInjection()
 
-                IAsyncResult asyncContext = readAResultAccessor(request);
-                if (asyncContext != null)
-                {
-                    // Flow here is for [Begin]GetResponse[Async] without a prior call to [Begin]GetRequestStream[Async].
+            IAsyncResult asyncContext = readAResultAccessor(request);
+            if (asyncContext != null)
+            {
+                // Flow here is for [Begin]GetResponse[Async] without a prior call to [Begin]GetRequestStream[Async].
 
-                    AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
-                    asyncCallbackModifier(asyncContext, callback.AsyncCallback);
-                }
-                else
-                {
-                    // Flow here is for [Begin]GetRequestStream[Async].
+                AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
+                asyncCallbackModifier(asyncContext, callback.AsyncCallback);
+            }
+            else
+            {
+                // Flow here is for [Begin]GetRequestStream[Async].
 
-                    asyncContext = writeAResultAccessor(request);
-                    AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
-                    asyncCallbackModifier(asyncContext, callback.AsyncCallback);
-                }
+                asyncContext = writeAResultAccessor(request);
+                AsyncCallbackWrapper callback = new AsyncCallbackWrapper(request, activity, asyncCallbackAccessor(asyncContext));
+                asyncCallbackModifier(asyncContext, callback.AsyncCallback);
+            }
 
-                InstrumentRequest(request, activity);
+            InstrumentRequest(request, activity);
 
-                // Only send start event to users who subscribed for it, but start activity anyway
-                if (this.IsEnabled(RequestStartName))
-                {
-                    this.Write(activity.OperationName + ".Start", new { Request = request });
-                }
-            }
+            activity.SetCustomProperty("request", request);
         }
 
-        private void RaiseResponseEvent(HttpWebRequest request, HttpWebResponse response)
+        private void RaiseResponseEvent(HttpWebResponse response)
         {
-            if (this.IsEnabled(RequestStopName))
+            Activity activity = Activity.Current;
+            if (activity == null)
             {
-                this.Write(RequestStopName, new { Request = request, Response = response });
+                return;
             }
+
+            activity.AddTag(SpanAttributeConstants.HttpStatusCodeKey, ((int)response.StatusCode).ToString());
+            activity.AddTag(SpanAttributeConstants.HttpStatusTextKey, response.StatusDescription);
+
+            activity.SetCustomProperty("response", response);
```

Comment:
I am looking for where activity.Stop is called. Can you point me to where we stop activity?

Reply:
It's called on line 205 in `ProcessResult` right after `RaiseResponseEvent` or `RaiseExceptionEvent` is fired.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/8e92f134b6e09b8fd02ba3808ca58b16b2913398/src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs#L205